howto/android/odm_vhal.md

```diff
@@ -0,0 +1,128 @@
+for your VHAL service to `/odm/etc/init`
+1. Disabling the Anbox Cloud VHAL by setting the `ro.anbox.automotive.vhal`
+property to `odm`. This will be done by writing that system property to the
+`/odm/etc/build.prop` file.
+
+In the following, we will name our addon `custom-vhal`.
+To override the VHAL implementation, we will use a
+[pre-start hook](https://anbox-cloud.io/docs/reference/hooks) which is executed
+before Android gets started.
+
+The directory layout for the addon is strict and must be:
+
+```
+custom-vhal/
+├── hooks
+│   └── pre-start
+├── manifest.yaml
+├── vhal
+└── vhal.rc
+```
+
+`vhal` is the VHAL implementation built in the previous section. If your VHAL
+implementation requires additional libraries, they must also be bundled with the
+addon and copied in the ODM partition.
+
+`vhal.rc` is an [init.rc file](https://android.googlesource.com/platform/system/core/+/main/init/README.md) for your VHAL service, such as:
+
+```
+service custom-vhal /odm/bin/hw/vhal
+    class hal
+    user vehicle_network
+    group system inet
+```
+
+`hooks/pre-start` must be an executable file which will handle the tasks outlined at the beginning of this section, such as the following bash script:
+
+```bash
+#!/bin/bash -x
+#
+# Copyright 2024 Canonical Ltd.  All rights reserved.
+#
+
+ANDROID_ODM_DIR="${ANBOX_DIR}/android-odm"
+
+if [ "${INSTANCE_TYPE}" != base ]; then
+    exit 0
+fi
+
+# Ensure all needed directories are created
+mkdir -p "${ANDROID_ODM_DIR}/etc/init"
+mkdir -p "${ANDROID_ODM_DIR}/bin/hw"
+
+# Copy the custom VHAL and the associated init file in /odm
+cp "${ADDON_DIR}/vhal" "${ANDROID_ODM_DIR}/bin/hw/"
+cp "${ADDON_DIR}/vhal.rc" "${ANDROID_ODM_DIR}/etc/init/"
+
+# Set the ro.anbox.anbox.vhal system property to 'odm'
+echo "ro.anbox.automotive.vhal=odm" >> "${ANDROID_ODM_DIR}/etc/build.prop"
+```
+
+The addon
+(`manifest.yaml`)[https://anbox-cloud.io/docs/reference/addon-manifest] file
+contains metadata:
+
+```yaml
+name: custom-vhal
+description: |
+  Addon replacing the Anbox Cloud VHAL with a custom implementation.
+```
+
+Once everything is in place, you can add the addon to your Anbox Cloud instance,
+with:
+
+```bash
+amc addon add custom-vhal ./custom-vhal
+```
+
+Please note that due to Snap strict confinement, the addon directory
+(`custom-vhal` here) must be located in the home directory of the user.
+
+## Use the addon
+
+The newly added `custom-vhal` addon must now be enabled to be used with
+Anbox Cloud applications.
+
+If you plan to always override the Anbox Cloud VHAL implementation in all
+applications, you can
+[enable the addon globally](https://anbox-cloud.io/docs/howto/addons/enable-globally):
+
+```bash
+amc config set application.addons custom-vhal
+```
+
+Otherwise, add it to your
+[application manifest](https://anbox-cloud.io/docs/reference/application-manifest):
+
+```yaml
+name: my-app
+addons:
+  - custom-vhal
+```
+
+## Additional information
```

Comment:
A very minor nit, pardon me, but I am trying to keep things as consistent as possible:

```suggestion
## Related information
```